src/pages/blog/optimizing-emotion.mdx

```diff
@@ -0,0 +1,293 @@
+  _Somewhere in your JSX..._
+</div>
+```tsx
+export const MyButton = ({
+  buttonHoverBorderColorOverride = "skyblue",
+  ...props
+}: { buttonHoverBorderColorOverride?: string } & React.DetailedHTMLProps<
+  React.ButtonHTMLAttributes<HTMLButtonElement>,
+  HTMLButtonElement
+>) => (
+  <button
+    {...props}
+    className={"my-button " + props.className}
+    style={{
+      "--button-hover-border-color": buttonHoverBorderColorOverride,
+      ...props.style,
+    }}
+  />
+);
+```
+<strong style={{ textAlign: "center", display: "block", paddingLeft: "1em", paddingRight: "1em" }}>
+  _Figure 8._ Passing state from React to raw CSS
+</strong>
+</Card>
+
+The reason Emotion is chosen for bigger component libraries is that it allows for high customizability with it's theme overrides model which is exposed by most component libraries that use it. This allows them to be highly overridable and configurable, but at the cost of a less perfomant solution to styling. If you are building a new component library, you may want to consider that you can sacrifice the extent of your customizability & configurability in exchange for a more performant solution. This would allow you to compete with bigger component libraries because while they would have the advantages that Emotion affords them, your library could have the performance advantage.
+
+If you really feel that you need what Emotion offers, there are still other alternatives in the runtime CSS-in-JS ecosystem. [Stitches](https://stitches.dev/) is a fantastic alternative of similar features to Emotion. It is a bit more opinionated than Emotion, and only allows for object-styles. However, this means that it doesn't have to support a custom string parser like Emotion does by relying on Stylis. Because of this, Stitches is smaller, and by offering less syntax options [it maintains better performance than Emotion in their open-sourced benchmarks](https://stitches.dev/docs/benchmarks).
+
+## Conclusion
+
+Emotion is a well engineered, unopinionated, and surprsingly performant ecosystem of tools that may offer the optimal implementation of a sensible solution for a complex problem, but it is not without it's drawbacks. It is important to understand the tradeoffs that you are making when you choose to use Emotion or any styling system in your application or library. It is also important to understand that there are many ways to optimize your use of Emotion, and the docs aren't always up-front about it and seem to focus more directly on the enhanced capabilities and developer experience than the potential drawbacks. If you come into a project with all this awareness in mind and are able to account for optimizations slightly earlier, than you can avoid [having to do a lot of refactoring later on if Emotion starts to feel like it's not meeting your needs](https://dev.to/srmagura/why-were-breaking-up-wiht-css-in-js-4g9b).
```

Comment:
Haha I was waiting for a link to this article!

Reply:
I intended to sneak it in earlier, but it felt the most sensible to put it here as it kind of seems like the overarching consequential risk of leaving Emotion code unopimized.